src/collection/components/CollectionFragment.tsx

```diff
@@ -0,0 +1,249 @@
+	index,
+	collection,
+	swapFragments,
+	updateFragmentProperty,
+	isOptionsMenuOpen,
+	setIsOptionsMenuOpen,
+	fragment,
+	reorderFragments,
+	updateCollection,
+}) => {
+	// Check whether the current fragment is the first and/or last fragment in collection
+	const isFirst = (index: number) => index === 0;
+	const isLast = (index: number) => index === collection.collection_fragments.length - 1;
+
+	// Render methods
+	const renderReorderButton = (fragmentId: number, direction: 'up' | 'down') => (
+		<Button
+			type="secondary"
+			icon={`chevron-${direction}`}
+			onClick={() => swapFragments(fragmentId, direction)}
+		/>
+	);
+
+	const renderForm = (fragment: any, index: number) => (
+		<Form>
+			<FormGroup label={`Tekstblok titel`} labelFor="title">
+				<TextInput
+					id="title"
+					type="text"
+					value={fragment.custom_title || ''}
+					placeholder="Titel"
+					onChange={(value: string) => updateFragmentProperty(value, 'custom_title', fragment.id)}
+				/>
+			</FormGroup>
+			<FormGroup label={`Tekstblok beschrijving`} labelFor={`beschrijving_${index}`}>
+				<>
+					<WYSIWYG
+						id={`beschrijving_${index}`}
+						data={fragment.custom_description || ''}
+						onChange={
+							((e: any) =>
+								updateFragmentProperty(
+									e.target.innerHTML,
+									'custom_description',
+									fragment.id
+								)) as any // TODO remove any type once components deploy is working again
+						}
+					/>
+				</>
+			</FormGroup>
+		</Form>
+	);
+
+	const onDuplicateFragment = (fragmentId: number) => {
+		setIsOptionsMenuOpen(null);
+		toastService('Fragment is succesvol gedupliceerd', TOAST_TYPE.SUCCESS);
+	};
+
+	const onMoveFragment = () => {
+		setIsOptionsMenuOpen(null);
+		toastService('Fragment is succesvol verplaatst', TOAST_TYPE.SUCCESS);
+	};
+
+	const onCopyFragmentToCollection = () => {
+		setIsOptionsMenuOpen(null);
+		toastService('Fragment is succesvol gekopiëerd naar collectie', TOAST_TYPE.SUCCESS);
+	};
+
+	const onMoveFragmentToCollection = () => {
+		setIsOptionsMenuOpen(null);
+		toastService('Fragment is succesvol verplaatst naar collectie', TOAST_TYPE.SUCCESS);
+	};
+
+	// Delete fragment from collection
+	const onDeleteFragment = (fragmentId: number) => {
+		setIsOptionsMenuOpen(null);
+
+		// Sort fragments by position
+		const orderedFragments = orderBy(
+			collection.collection_fragments.filter(
+				(fragment: Avo.Collection.Fragment) => fragment.id !== fragmentId
+			),
+			['position'],
+			['asc']
+		);
+
+		const updatedFragmentIds = (collection.collection_fragment_ids || []).filter((id: number) => {
+			return id !== fragmentId;
+		});
+
+		const positionedFragments = reorderFragments(orderedFragments);
+
+		updateCollection({
+			...collection,
+			collection_fragments: positionedFragments,
+			collection_fragment_ids: updatedFragmentIds,
+		});
+
+		toastService('Fragment is succesvol verwijderd', TOAST_TYPE.SUCCESS);
+	};
+
+	const renderCollectionFragment = (itemMeta: any) => (
```

Comment:
Ik vermoed dat dit in een toekomstige PR gebruikt zal worden eens je de toggle component hebt?

Reply:
Ja.